diff --git a/GenerateCLP/CMakeLists.txt b/GenerateCLP/CMakeLists.txt
index 75fabef..ec13767 100644
--- a/GenerateCLP/CMakeLists.txt
+++ b/GenerateCLP/CMakeLists.txt
@@ -137,12 +137,12 @@ endif()
 # Install support files
 # --------------------------------------------------------------------------
 if(NOT ${PROJECT_NAME}_INSTALL_NO_DEVELOPMENT)
-  install(FILES ${GenerateCLP_BINARY_DIR}/GenerateCLPConfig.cmake_install
+  install(FILES ${GenerateCLP_BINARY_DIR}/cmake_install.cmake
     DESTINATION lib/GenerateCLP
     COMPONENT Development
     RENAME GenerateCLPConfig.cmake
     )
-  install(FILES ${GenerateCLP_BINARY_DIR}/UseGenerateCLP.cmake_install
+  install(FILES ${GenerateCLP_BINARY_DIR}/cmake_install.cmake
     DESTINATION lib/GenerateCLP
     COMPONENT Development
     RENAME UseGenerateCLP.cmake
